src/test/java/io/confluent/connect/jdbc/source/integration/MSSQLDateTimeIT.java

```diff
@@ -0,0 +1,314 @@
+                ZonedDateTime.of(2016, 12, 8, 19, 34, 56, 2000000, utc).toInstant()
+        );
+
+        writeSingleRowWithTimestamp(t, false, true);
+
+        Thread.sleep(Duration.ofSeconds(30).toMillis());
+        for (String topic: KAFKA_TOPICS) {
+            ConsumerRecords<byte[], byte[]> records = connect.kafka().consume(
+                    NUM_RECORDS_PRODUCED,
+                    CONSUME_MAX_DURATION_MS,
+                    topic);
+            //Assert that records in topic == NUM_RECORDS_PRODUCED
+            assertEquals(NUM_RECORDS_PRODUCED, records.count());
+
+        }
+        // clean up table
+        deleteTable();
+    }
+
+    private Map<String, String> configProperties(boolean timestampModeOnly) {
+        // Create a hashmap to setup source connector config properties
+        Map<String, String> props = new HashMap<>();
+        props.put(CONNECTOR_CLASS_CONFIG, "JdbcSourceConnector");
+        props.put(CONNECTION_URL_CONFIG, MSSQL_URL);
+        props.put(CONNECTION_USER_CONFIG, "sa");
+        props.put(CONNECTION_PASSWORD_CONFIG, "reallyStrongPwd123");
+        props.put(TABLE_WHITELIST_CONFIG, MSSQL_Table);
+
+        if (timestampModeOnly) {
+            props.put(MODE_CONFIG, MODE_TIMESTAMP);
+        } else {
+            props.put(MODE_CONFIG, MODE_TIMESTAMP_INCREMENTING);
+            props.put(INCREMENTING_COLUMN_NAME_CONFIG, "ict");
+        }
+
+        props.put(TIMESTAMP_COLUMN_NAME_CONFIG, "start_time");
+        props.put(TOPIC_PREFIX_CONFIG, "test-");
+        props.put(POLL_INTERVAL_MS_CONFIG, "30");
+        props.put(TASKS_MAX_CONFIG, Integer.toString(TASKS_MAX));
+        props.put(KEY_CONVERTER_CLASS_CONFIG, StringConverter.class.getName());
+        props.put(VALUE_CONVERTER_CLASS_CONFIG, StringConverter.class.getName());
+        props.put(CONFLUENT_TOPIC_BOOTSTRAP_SERVERS_CONFIG, connect.kafka().bootstrapServers());
+        props.put(CONFLUENT_TOPIC_REPLICATION_FACTOR_CONFIG, "1");
+
+        return props;
+    }
+
+    private void createDateTimeTable(
+            String timestampColType,
+            String regularColType,
+            boolean timestampModeOnly
+    ) throws Exception {
+        try {
+            Statement stmt = connection.createStatement();
+            StringBuilder builder = new StringBuilder();
+            builder.append("CREATE TABLE " + MSSQL_Table )
+                    .append("(start_time " + timestampColType +  " not NULL, " );
+            if (!timestampModeOnly) builder.append(" ict  int not NULL, ");
+            builder.append(" record " + regularColType +  " )");
+
+            String sql = builder.toString();
+
+            stmt.executeUpdate(sql);
+            log.info("Successfully created table");
+        } catch (Exception ex) {
+            log.error("Could not create table");
+            throw ex;
+        }
+    }
+
+    private void writeSingleRowWithTimestamp(
+            Timestamp t,
+            boolean stringRecord,
```

Comment:
These boolean arguments are kind of un-intuitive, and their implementation is a bit awkward. Maybe more legible, and still good code re-use, to just accept the entire INSERT statement as a String. It may feel like you need to abstract out the actual sql statements, but since we are testing sql implementation details, I think it's fine to surface the call itself.